crypto/signatures_test.go

```diff
@@ -2,58 +2,171 @@ package crypto
 
 import (
 	"crypto/rand"
+	"errors"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
```

Comment:
I'm beginning to think you and @lukechampine have Stockholm Syndrome. There's no other explanation for your preference for go's built in testing package, which to me is horrible. I've added more details on the long reply to Luke, but tl; dr - I think we need `mock`, I prefer `assert/require` but I defer to your preferences.